orange/src/main/java/com/hartwig/hmftools/orange/algo/OrangeAlgo.java

```diff
@@ -252,30 +252,62 @@ public OrangeRecord run(@NotNull OrangeConfig config) throws Exception
         }
 
         boolean hasRefSample = config.wgsRefConfig() != null && config.wgsRefConfig().referenceSampleId() != null;
-        OrangeRecord report = ImmutableOrangeRecord.builder()
-                .sampleId(config.tumorSampleId())
-                .samplingDate(config.samplingDate())
-                .experimentType(config.experimentType())
-                .configuredPrimaryTumor(ConversionUtil.mapToIterable(configuredPrimaryTumor, OrangeConversion::convert))
-                .refGenomeVersion(config.refGenomeVersion())
-                .pipelineVersion(pipelineVersion)
-                .refSample(refSample)
-                .tumorSample(tumorSample)
-                .germlineMVLHPerGene(mvlhPerGene)
-                .purple(purple)
-                .linx(linx)
-                .wildTypeGenes(wildTypeGenes)
-                .isofox(isofox)
-                .lilac(lilac != null ? OrangeConversion.convert(lilac, hasRefSample, config.rnaConfig() != null) : null)
-                .immuneEscape(immuneEscape)
-                .virusInterpreter(virusInterpreter != null ? VirusInterpreter.interpret(virusInterpreter) : null)
-                .chord(chord != null ? OrangeConversion.convert(chord) : null)
-                .cuppa(cuppa)
-                .peach(ConversionUtil.mapToIterable(peach, OrangeConversion::convert))
-                .sigAllocations(SigsInterpreter.interpret(sigAllocations, etiologyPerSignature))
-                .cohortEvaluations(evaluateCohortPercentiles(config, purple))
-                .plots(buildPlots(config))
-                .build();
+        OrangeRecord report;
+        if (peach != null)
```

Comment:
having the null check up here introduced a lot of duplication, let's try to narrow the scope.